themes/src/main/resources/theme/base/login/resources/js/authChecker.js

```diff
@@ -32,7 +32,7 @@ function checkAuthState(authSessionId, tabId, loginRestartUrl) {
   // Attempt to parse the auth state as JSON.
   let authState;
   try {
-    authState = JSON.parse(authStateRaw);
+    authState = JSON.parse(decodeURIComponent(authStateRaw));
```

Comment:
Why must this now be decoded?

Reply:
Because this is a JSON value, and not something that fits well into a cookie value un-encoded. I did urlencode as that's easy on the JS side, ideally this would probably be Base64 encoded to match other cookie types, but that would be much harder to parse on the JS side as it would require including a library.

Reply:
> but that would be much harder to parse on the JS side as it would require including a library.

We are able to do this [in plain JS](https://developer.mozilla.org/en-US/docs/Glossary/Base64#javascript_support) these days.

Reply:
There's this https://developer.mozilla.org/en-US/docs/Glossary/Base64#the_unicode_problem, so I'm going to leave it as url-encoded, that works fine I reckon

Reply:
Very well :+1: